pkgs/os-specific/linux/cryptsetup/default.nix

```diff
@@ -16,6 +16,18 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     patchShebangs tests
+
+    # O_DIRECT is filesystem dependent and fails in a sandbox (on tmpfs)
+    # and on several filesystem types (btrfs, zfs) without sandboxing.
+    # Remove it, see discussion in #46151
+    substituteInPlace tests/unit-utils-io.c --replace "| O_DIRECT" ""
+
+    # Disable 4 test cases that still fail in a sandbox
+    substituteInPlace tests/blockwise-compat \
+      --replace 'RUN "F" $1 read_buffer $((BSIZE-1))' "" \
+      --replace 'RUN "F" $1 read_buffer $((BSIZE+1))' "" \
+      --replace 'RUN "F" $1 write_buffer $((BSIZE-1))' "" \
+      --replace 'RUN "F" $1 write_buffer $((BSIZE+1))' ""
```

Comment:
Maybe a path for this one? Otherwise it is hard to see, if this file needs an update or not (if new tests are added or fail in the sandbox).

Reply:
done.